### ISSUE TITLE
ResolvedRequestObject should include parsed Relying Party name or it's certificate chain

### ISSUE THREAD

**Issue by ydanneg** [eu-digital-identity-wallet/eudi-lib-jvm-siop-openid4vp-kt#235]
According to [ARF](https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/blob/main/docs/arf.md#757informing-the-user-about-relying-party-authentication-outcomes) Wallet must show Relying Party name taken from RP Instance's Certificate.

Issue: Currently there is no way to get access to RP certs when using AuthorizationRequestResolver.

PS. ARF now only specifies it for 'ISO/IEC 18013-5' only, 'SD-JWT' part is not specified yet, but I think it should work the same way.

**Comment by ydanneg** [comment]
> Hi @ydanneg
> 
> I am not sure that I follow, what is missing.
> 
> Assuming that you refer to a RP/Verifier that uses`client_id_scheme` equal to `x5c_san_dns`, then the name that you mention is the `client_id` which is included in the resolved authentication, provided that the Verifier/RP cert is trusted by the wallet.
> 
> The later is handled in
> 
> https://github.com/eu-digital-identity-wallet/eudi-lib-jvm-siop-openid4vp-kt/blob/49f5d279f466eb44a32738efa35b8a57f620b5a0/src/main/kotlin/eu/europa/ec/eudi/openid4vp/internal/request/RequestAuthenticator.kt#L102
> 
> Can you please elaborate?

Hi @babisRoutis 

Saying Verifier name i mean Verifier's legal name.

`client_id` will  contains dnsName (used for protocol's internal validations).
Legal name is a "representative" name of a Verifier that can be shown in Wallet UI.

`
The Relying Party Instance certificate SHALL contain the legal name for the Relying Party in the subject's Common Name, which the Wallet Instance can show to the User when asking for User approval, see section 7.6.1. The Relying Party Instance certificate MAY contain a Relying Party name extension that further defines the Relying Party service that, if existent, will also be shown to the User.
`
see https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/blob/main/docs/arf.md#7582-relying-party-instance-certificate-format

As a library user I would like to get this info from `Resolution` and show it in UI after calling `eu.europa.ec.eudi.openid4vp.AuthorizationRequestResolver#resolveRequestUri`

Thanks in advance,
Gennadi

**Comment by babisRoutis** [comment]
@ydanneg 
Now it is clear :+1: 

I guess that we can introduce a new sealed interface especially for `ResolvedRequestObject` that will replace `clientId`

Something like
```kotlin
sealed interface Client {
    data class Preregistered(val clientId: String, val name: String) : Client
    data class RedirectUri(val clientId: URI) : Client
    data class X509SanDns(val clientId: String, val cert: X509Certificate) : Client
    data class X509SanUri(val clientId: URI, val cert: X509Certificate) : Client
}

fun AthClient.legalName() : String = when(this) {
  is Preregistered -> name
  is RedirectUri -> "N/A" // This is not supported in ARF
  is X509SanDns -> extractCnFrom(cert)
  is X509SanUri -> extractCnFrom(cert)
}


sealed interface ResolvedRequestObject : Serializable {

    val client: Client
}
```

**Comment by dzarras** [comment]
@ydanneg Thanks for reporting this. We are already working on a fix.

**Comment by babisRoutis** [comment]
@ydanneg FYI released [v0.3.5](https://github.com/eu-digital-identity-wallet/eudi-lib-jvm-siop-openid4vp-kt/releases/tag/v0.3.5)

Thanks again for the valuable contribution
